snippets/promisify.md

```diff
@@ -0,0 +1,15 @@
+### Promisify
+
+Creates a promisified version of the given callback-style function. In Node 8+, you
+can use [`util.promisify`](https://nodejs.org/api/util.html#util_util_promisify_original)
+
+```js
+const promisify = func =>
+  (...args) =>
+    new Promise((resolve, reject) =>
+      func(...args, (err, result) =>
+        err ? reject(err) : resolve(result));
+    )
+// const stat = promisify(fs.stat)
+// stat('foo.txt') -> Promise resolves if `foo.txt` exists, otherwise rejects
```

Comment:
Can we add an example that could run in the browser without Node? I'd prefer that, so that people can copy paste and test right away. If not, it's ok.

Reply:
Humm I've been thinking a callback-style operation in the browser, but can't think of any. Most of them already returns a promise. Do you know any :thinking: ?

Reply:
A timeout-based one (like a function with a timeout that returns a value) maybe?

Reply:
`const delay = (d, cb) => setTimeout(cb, d)`?

Reply:
`setTimeout` has the callback in the first argument of the function, instead of the last... :disappointed:

Reply:
I think setTimeout is a bit ugly for this use case, what do you think @Chalarangelo ?

```js
promisify((d, cb) => setTimeout(cb, d))
```

Reply:
I don't hate it to be honest, but I might be in the minority. @darrenscerri what do you think?

Reply:
I think the idea is to promisify any callback-style function, which do not necessarily need to be native APIs. I guess that's fine and drives the point across.

Reply:
Alright, let's stick with that example then, at least for now. If anybody else has any better ideas later down the line, that's why we have PRs! 😉